### PR TITLE
Feature/cb2 3409 test result vehicle section

### PR DESCRIPTION
**Pull request by danielchildsbjss**
## CB2-3409 - Added Test results "Vehicle" section

_Added additional section to Test results page. Added ability to concatenate multiple fields to suit the data. CSS changes to accommodate new fields._
[CB2-3409](https://jira.dvsacloud.uk/browse/CB2-3409)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/forms/components/view-combination/view-combination.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  separator: string = ' ';
+  label?: string;
+
+  constructor() {
+    this.formNode = <FormNode>{};
+    this.formGroup = <FormGroup>{};
+  }
+
+  ngOnInit(): void {
+    console.log(this.formNode)
```
[the remark this answers is not in the document]
Comment (kpearmanbjss) [reply src/app/forms/components/view-combination/view-combination.component.ts]: I think that was left behind from what Tom was working on, i think i've taken it out now!

File: src/app/forms/components/dynamic-form-group/dynamic-form-group.component.html [hunk 2]
```diff
@@ -10,21 +11,49 @@
       </ng-container>
 
       <ng-container *ngSwitchCase="formNodeViewTypes.GROUP">
-        <ng-template [ngTemplateOutlet]="nodeTemplateRef" [ngTemplateOutletContext]="{ form: form.controls[control.key] }"></ng-template>
+        <ng-template [ngTemplateOutlet]="childNode" [ngTemplateOutletContext]="{ form: form.controls[control.key] }"></ng-template>
       </ng-container>
 
       <ng-container *ngSwitchCase="formNodeViewTypes.ARRAY">
         <ng-template [ngTemplateOutlet]="nodeTemplateRef" [ngTemplateOutletContext]="{ form: form.controls[control.key] }"></ng-template>
```
[the remark this answers is not in the document]
Comment (mcadenas-bjss) [reply src/app/forms/components/dynamic-form-group/dynamic-form-group.component.html:18]: Arrays are handled by the same (`nodeTemplateRef`) template as it goes by keys

File: src/app/forms/components/view-combination/view-combination.component.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  ngOnInit(): void {
+    const options = <FormNodeCombinationOptions>(this.formNode.options);
+    this.leftComponent = this.findComponentByName(options.leftComponentName, this.formGroup);
+    this.rightComponent = this.findComponentByName(options.rightComponentName, this.formGroup)
+    this.separator = options.separator;
+    this.label = this.formNode.label;
+  }
+
```
Comment (premghinde) [inline src/app/forms/components/view-combination/view-combination.component.ts:39]: Excessive whiltespace should be caught by linting

File: src/app/models/country-of-registration.enum.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+export enum CountryOfRegistration {
+  GreatBritainandNorthernIreland_GB = 'gb'
+  // 'Alderney - GBA' = 'gba',
```
Comment (premghinde) [inline src/app/models/country-of-registration.enum.ts:3]: Any reason to keep commented out country codes?

**Review by mcadenas-bjss — DISMISSED** [review]

**Review by gjulien-bjss — DISMISSED** [review]

File: src/app/forms/services/dynamic-form.service.ts [hunk 5]
```diff
@@ -20,13 +21,12 @@ export class DynamicFormService {
...
     f?.children.forEach((child) => {
-      const { name, type, value, validators, disabled, readonly } = child;
-      let control: any;
-
+      const { name, type, value, validators, disabled } = child;
+      let control;
       if (FormNodeTypes.CONTROL !== type) {
         control = this.createForm(child, d[name]);
       } else {
-        control = new CustomFormControl({ ...child, readonly: !!readonly }, { value, disabled: !!disabled });
```
[the remark this answers is not in the document]
Comment (naathanbrown) [reply src/app/forms/services/dynamic-form.service.ts:29]: @mcadenas-bjss can you confirm these need to go too, I think so but want to be clear as you mainly wrote that new section

[the remark this answers is not in the document]
Comment (mcadenas-bjss) [reply src/app/forms/services/dynamic-form.service.ts:29]: that's true, didn't see that.

**Review by naathanbrown — APPROVED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Review by mcadenas-bjss — APPROVED** [review]

**Pull request merged**